src/MongoDB.Driver.Core/Core/Operations/ChangeStreamCursor.cs

```diff
@@ -160,7 +159,8 @@ private IEnumerable<TDocument> DeserializeDocuments(IEnumerable<RawBsonDocument>
 
             if (lastRawDocument != null)
             {
-                _resumeToken = lastRawDocument["_id"].DeepClone().AsBsonDocument;
+                _changeStreamOperation.ResumeAfter = lastRawDocument["_id"].DeepClone().AsBsonDocument;
+                _changeStreamOperation.StartAtOperationTime = null;
```

Comment:
Resume token is now stored in the operation.

When adding a resume token we clear out the startAtOperationTime because the resume token supercedes it.